java/java.hints/src/org/netbeans/modules/java/hints/perf/Tiny.java

```diff
@@ -337,11 +353,24 @@ public static ErrorDescription collectionsToArray(HintContext ctx) {
         Fix[] fixes;
 
         if (pureMemberSelect) {
-            String fixDisplayName = NbBundle.getMessage(Tiny.class, "FIX_Tiny_collectionsToArray");
 
-            fixes = new Fix[] {
-                JavaFixUtilities.rewriteFix(ctx, fixDisplayName, ctx.getPath(), "$collection.toArray(new $clazz[$collection.size()])")
-            };
+            SourceVersion version = ctx.getInfo().getSourceVersion();
+            TreePath type = ctx.getVariables().get("$clazz");
+            String typeName = type.getLeaf().toString();
+            
+            if (Integer.parseInt(version.name().substring(version.name().indexOf('_')+1)) >= 11) {
```

Comment:
Use of `SourceVersion`/`CompilationInfo.getSourceVersion()` is correct, but `SourceVersion` is `Comparable`, so can be compared without integer tricks. Please see method `unnecessaryTempFromString` in this class for an example.

Reply:
can i compare against 11 if NB is running on <11 without nb-javac? SourceVersion is part of the JDK if i see this correctly - the enum would not be there.

Reply:
unnecessaryTempFromString compares against 5 which doesn't have the same problem since 8 is minimum required JDK version for NB.

Reply:
Right again, but I think I'd still somewhat prefer to compare SourceVersions than play integer tricks. I'd try to create a static final field, initialize to SourceVersion.RELEASE_11 if it exists (`SourceVersion.valueOf("RELEASE_11")`), or null. Hopefully easier to find when we can depend on the new API.

Thanks.

Reply:
I implemented it here locally, but i gotta say it still looks very hacky. Esp the 
`MODEL_VERSION_11 != null && MODEL_VERSION_11.compareTo(version) >= 0
The main issue i see is that it will spread, other hints would have to keep doing the same thing, there is no nice way to make utility method out of this, to keep the ugly part in one place.

I am glad that `Runtime.Version` uses ints to represent the feature version since this shows that enums are a bad idea for things like that (many other projects ran into the same trap).

what we actually want to write is
```
version.supports(11)
```
or even better
```
version.feature() >= 11
```
Thats it, no time traveling needed to figure out potential future enums.

what would you say to a reusable utility like:
 ```
    public static boolean supports(SourceVersion model, int featureVersionToCheck) {
        return model.ordinal() >= featureVersionToCheck;
    }
```
This should be forward compatible for a long time, until JDK hopefully adds `version.feature()` to make it compatible with their own Runtime.version().feature() before breaking the enum ordering.